src/components/Providers/SystemIOProviderDesktop.tsx

```diff
@@ -284,6 +286,15 @@ export function SystemIOMachineLogicListenerDesktop() {
     }
   )
 
+  // ZOOKEEPER BEHAVIOR EXCEPTION
+  // Only fires on state changes, to deal with Zookeeper control.
+  const systemIOState = useSelector(systemIOActor, (actor) => actor.value)
+  useEffect(() => {
+    if (systemIOState !== 'idle') return
+    void kclManager.executeCode()
+    kclManager.mlEphantManagerMachineBulkManipulatingFileSystem = false
+  }, [systemIOState])
```

Comment:
This is dangerous because it runs at every location within the application and it runs when it should not run. I posted a video in our slack thread outlining the issues.